--- GNUmakefile.am
+++ GNUmakefile.am
@@ -18,8 +18,6 @@ wvhdr_HEADERS = wv.h
 
 bin_SCRIPTS = wvCleanLatex wvHtml wvDocBook wvAbw wvMime wvPS wvDVI wvLatex wvPDF wvText wvWml
 
-bin_PROGRAMS = wvWare wvRTF wvConvert wvVersion wvSummary
-
 lib_LTLIBRARIES = libwv.la
 
 GETOPT = @LIBOBJS@
--- wv.h
+++ wv.h
@@ -10,6 +10,9 @@
 extern "C" {
 #endif
 
+int printfRedirected(const char* __fmt, ...);
+#define printf printfRedirect
+
 /* The structure below is used to refer to a wvStream.  Usually,
  * kind = GSF_STREAM,
  * but if we can't open a file using LibGSF, we fall back to the old file-based
