From cc61f0ae52b642d8e6560ddaed26bcf3fd5eae01 Mon Sep 17 00:00:00 2001
From: Andreas Stefl <stefl.andreas@gmail.com>
Date: Wed, 18 Sep 2024 15:56:00 +0200
Subject: [PATCH 4/4] gcc14 build fix

---
 wvConfig.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wvConfig.c b/wvConfig.c
index e44f4f6..840fa93 100644
--- a/wvConfig.c
+++ b/wvConfig.c
@@ -158,7 +158,7 @@ wvReleaseStateData (state_data * data)
 
 
 static void
-exstartElement (void *userData, const char *name, const char **atts)
+exstartElement (void *userData, const unsigned char *name, const unsigned char **atts)
 {
     unsigned int token_type;
     expand_data *mydata = (expand_data *) userData;
@@ -3371,7 +3371,7 @@ wvendElement (void *userData, const XML_Char *name)
 }
 
 static void
-exendElement (void *userData, const char *name)
+exendElement (void *userData, const unsigned char *name)
 {
     /*
        expand_data *mydata = (expand_data *)userData;
-- 
2.39.3 (Apple Git-146)

